File: api/v2010/incoming_phone_number/read-example-2/read-example-2.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

incoming_phone_numbers = @client.incoming_phone_numbers
                                .list(phone_number: '+14158675310')

incoming_phone_numbers.each do |record|
  puts record.sid
end

File: api/v2010/outgoing_caller_id/read-example-2/read-example-2.5.x.cs
// Install the C# / .NET helper library from twilio.com/docs/csharp/install

using System;
using Twilio;
using Twilio.Rest.Api.V2010.Account;


class Program 
{
    static void Main(string[] args)
    {
        // Find your Account Sid and Token at twilio.com/console
        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
        const string authToken = "your_auth_token";

        TwilioClient.Init(accountSid, authToken);

        var outgoingCallerIds = OutgoingCallerIdResource.Read(
            phoneNumber: new Twilio.Types.PhoneNumber("+14158675310")
        );

        foreach(var record in outgoingCallerIds)
        {
           Console.WriteLine(record.Sid);
        }
    }
}
